src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs

```diff
@@ -9,41 +9,44 @@ namespace System.IO.Enumeration
     /// </summary>
     public unsafe ref struct FileSystemEntry
     {
-        // TODO: Unix implementation https://github.com/dotnet/corefx/issues/26715
-        // Inital implementation is naive and not optimized.
-
         internal static void Initialize(
             ref FileSystemEntry entry,
             Interop.Sys.DirectoryEntry directoryEntry,
             bool isDirectory,
             ReadOnlySpan<char> directory,
             string rootDirectory,
-            string originalRootDirectory)
+            string originalRootDirectory,
+            Span<char> pathBuffer)
         {
             entry._directoryEntry = directoryEntry;
             entry._isDirectory = isDirectory;
             entry.Directory = directory;
             entry.RootDirectory = rootDirectory;
             entry.OriginalRootDirectory = originalRootDirectory;
+            entry._pathBuffer = pathBuffer;
+            FileStatus.Initialize(ref entry._status, isDirectory);
         }
 
         internal Interop.Sys.DirectoryEntry _directoryEntry;
-        private FileSystemInfo _info;
+        private FileStatus _status;
         private bool _isDirectory;
```

Comment:
why do oyu need an _isdirectory field when it's on _status.Isdirectory?

Reply:
I don't have a property for it on FileStatus, but I can make it such.